interfaces/apparmor/backend_test.go

```diff
@@ -492,7 +495,362 @@ func (s *backendSuite) TestSetupHostSnapConfineApparmorForReexecWritesNew(c *C)
 	})
 
 	// snap-confine.d was created
-	_, err = os.Stat(dirs.SnapAppArmorConfineDir)
+	_, err = os.Stat(dirs.SnapConfineAppArmorDir)
 	c.Check(err, IsNil)
 
 }
+
+func (s *backendSuite) TestIsHomeUsingNFS(c *C) {
+	cases := []struct {
+		mountinfo, fstab string
+		nfs              bool
+		errorPattern     string
+	}{
+		// Errors from parsing mountinfo and fstab are propagated.
+		{
+			mountinfo:    "bad syntax",
+			errorPattern: "cannot parse /proc/self/mountinfo, .*",
+		},
+		{
+			fstab:        "bad syntax",
+			errorPattern: "cannot parse /etc/fstab, .*",
+		},
+		// NFSv4 currently mounted at /home/zyga/nfs is recognized.
+		{
+			mountinfo: "680 27 0:59 / /home/zyga/nfs rw,relatime shared:478 - nfs4 localhost:/srv/nfs rw,vers=4.2,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=127.0.0.1,local_lock=none,addr=127.0.0.1",
+			nfs:       true,
+		},
+		// NFSv4 currently mounted at /mnt/nfs is ignored (not in $HOME).
+		{
+			mountinfo: "680 27 0:59 / /mnt/nfs rw,relatime shared:478 - nfs4 localhost:/srv/nfs rw,vers=4.2,rsize=524288,wsize=524288,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=127.0.0.1,local_lock=none,addr=127.0.0.1",
+		},
+		// NFSv4 that may be mounted at /home/zyga/nfs is recognized.
+		{
+			fstab: "localhost:/srv/nfs /home/zyga/nfs nfs defaults 0 0",
+			nfs:   true,
+		},
```

Comment:
fstab entry has 'nfs', but comment is NFSv4 here and the test below. I like these tests, we probably want tests for nfs4 and (at least spot checks for) nfs.

Reply:
This is how it works actually. You get NFSv4 by default but you can still specify NFSv3 by using nfsvers=3. I'll expand tests to include NFSv3 as well.